metricflow/test/model/validations/test_reserved_keywords.py

```diff
@@ -0,0 +1,84 @@
+    return copy.deepcopy(simple_model__pre_transforms)
+
+
+def test_no_reserved_keywords(simple_model__pre_transforms: UserConfiguredModel) -> None:  # noqa: D
+    issues = ReservedKeywordsRule.validate_model(simple_model__pre_transforms)
+    assert len(issues) == 0
+
+
+def test_reserved_keywords_in_sql_table(simple_model__pre_transforms: UserConfiguredModel):  # noqa: D
+    model = copied_model(simple_model__pre_transforms)
+    (data_source_with_sql_table, _index) = find_data_source_with(
+        model=model, function=lambda data_source: data_source.sql_table is not None
+    )
+    data_source_with_sql_table.sql_table = f"{random_keyword()}.{random_keyword()}"
+
+    issues = ReservedKeywordsRule.validate_model(model)
+    assert len(issues) == 1
+    assert issues[0].level == ValidationIssueLevel.ERROR
+
+
+def test_reserved_keywords_in_dimensions(simple_model__pre_transforms: UserConfiguredModel):  # noqa: D
+    model = copied_model(simple_model__pre_transforms)
+    (data_source, _index) = find_data_source_with(
+        model=model, function=lambda data_source: len(data_source.dimensions) > 0
+    )
+    dimension = data_source.dimensions[0]
+    dimension.name = random_keyword()
+
+    issues = ReservedKeywordsRule.validate_model(model)
+    assert len(issues) == 1
+    assert issues[0].level == ValidationIssueLevel.ERROR
+
+
+def test_reserved_keywords_in_measures(simple_model__pre_transforms: UserConfiguredModel):  # noqa: D
+    model = copied_model(simple_model__pre_transforms)
+    (data_source, _index) = find_data_source_with(
+        model=model, function=lambda data_source: len(data_source.measures) > 0
+    )
+    measure = data_source.measures[0]
+    measure.name = random_keyword()
+
+    issues = ReservedKeywordsRule.validate_model(model)
+    assert len(issues) == 1
+    assert issues[0].level == ValidationIssueLevel.ERROR
+
+
+def test_reserved_keywords_in_identifiers(simple_model__pre_transforms: UserConfiguredModel):  # noqa: D
+    model = copied_model(simple_model__pre_transforms)
+    (data_source, _index) = find_data_source_with(
+        model=model, function=lambda data_source: len(data_source.identifiers) > 0
+    )
+    identifier = data_source.identifiers[0]
+    identifier.name = random_keyword()
+    identifier.identifiers = []
+
+    issues = ReservedKeywordsRule.validate_model(model)
+    assert len(issues) == 1
+    assert issues[0].level == ValidationIssueLevel.ERROR
+
+    identifier.name = "a_good_name"
+    identifier.identifiers = [
+        CompositeSubIdentifier(name=random_keyword()),
+        CompositeSubIdentifier(name=random_keyword()),
+        CompositeSubIdentifier(expr="SELECT TRUE AS col1"),  # shouldn't error
+    ]
+
+    issues = ReservedKeywordsRule.validate_model(model)
+    assert len(issues) == 2
+    assert issues[0].level == ValidationIssueLevel.ERROR
+    assert issues[1].level == ValidationIssueLevel.ERROR
```

Comment:
I'd prefer to have this in its own test case. It's a little confusing as written because the mutation puts it back in a good state.